### PR TITLE
feat: add antenna endpoints (GET/PUT /api/antennas)

### LINKED ISSUE #83: Criar endpoints de Antenas
### Objetivo

Permitir que o frontend consulte e atualize antenas. Todas as rotas devem ser restritas **apenas para administradores**.

O único trabalho do backend é validar a autenticação, os dados e encaminhar a requisição para o gateway. Não é necessário armazenar nenhuma informação no banco de dados.

### Endpoints:

---

`GET /api/antennas`
Retorna todas as antenas.

### Response

```json
[
  {
    "id": "guid",
    "number": 1,
    "status": "On",
    "sensibility": 80,
    "power": 30
  }
]
```

---

`GET /api/antennas/{id}`

Retorna uma antena específica.

---

`PUT /api/antennas/{id}`

Atualiza configurações da antena.

### Request

```json
{
  "status": "On",
  "sensibility": 80,
  "power": 30
}
```

---

### Regras
- Se status não vier, assumir Off
- Sensibility e Power podem ser nulos
- Validar limites mínimos e máximos

### Critérios de aceitação:
- Gateway é chamado corretamente
- Endpoints documentados no Swagger
- Testes de integração validam autenticação

### PR DESCRIPTION
**Pull request by Bepaiva**
## Summary

- Add three admin-only antenna endpoints that proxy to the external gateway (no DB storage):
  - `GET /api/antennas` — list all antennas
  - `GET /api/antennas/{id}` — get one antenna by ID
  - `PUT /api/antennas/{id}` — update antenna settings (status, sensibility, power)
- Backend validates auth and input, then forwards to the gateway via a typed `IGatewayClient` HTTP client
- Validation: `status` defaults to `"Off"` if omitted; `sensibility` and `power` must be in `[0, 100]`; `status` must be `"On"`, `"Off"`, or null

## Architecture

```
AntennaController → IAntennaService → IGatewayClient → Gateway
```

Gateway base URL is configurable via `Gateway:BaseUrl` in `appsettings.json`.

## Test Plan

- [x] 12 unit tests for `AntennaController` (all exception paths: 200, 400, 404, 502, 500)
- [x] 12 unit tests for `AntennaService` (delegation, null-status default, boundary values 0/100, invalid values -1/101)
- [x] 7 integration tests validating auth (401 unauthenticated, 403 Customer, 200 Admin) on all three endpoints
- [x] 156 unit tests passing, 161 integration tests passing

Closes #83

### CODE REVIEW

File: Backend.API/appsettings.json [hunk 1]
```diff
@@ -2,6 +2,9 @@
   "ConnectionStrings": {
     "Default": "Host=localhost;Port=5432;Database=mydatabase;Username=myuser;Password=mypassword"
   },
+  "Gateway": {
+    "BaseUrl": "http://localhost:8080"
+  },
```
Comment (BrunoPanizzi) [inline Backend.API/appsettings.json:7]: Tendo em vista que o gateway vai ser uma aplicação rodando em uma máquina local e não vai ter um endereço estático, acredito que uma variável de ambiente não é o melhor lugar para guardar o endpoint.

Exemplo, o servidor local reinicia, e com isso o endereço IP do gateway muda. Para que o backend mantenha a conexão, seria necessário alterar a variável no console da aws e realizar outro deploy para que as alterações sejam refletidas.

Podemos falar mais sobre isso na aula amanhã

File: Backend.API/Features/Antennas/Dtos/AntennaDto.cs
Comment (BrunoPanizzi) [inline Backend.API/Features/Antennas/Dtos/AntennaDto.cs:1]: Esses campos não batem com o gateway.

```csharp
// gateway/Gateway.API/Model/AntennaDto.cs
public sealed record AntennaResponse(
    ushort Port,
    double Power,
    double Sensitivity,
    bool Connected
);
```

File: Backend.API/Features/Antennas/Dtos/UpdateAntennaDto.cs
Comment (BrunoPanizzi) [inline Backend.API/Features/Antennas/Dtos/UpdateAntennaDto.cs:1]: Também não bate com o gateway

```csharp
// gateway/Gateway.API/Model/AntennaDto.cs
public sealed record AntennaUpdateRequest(
    double? Power,
    double? Sensitivity
);
```

File: Backend.API/Features/Antennas/GatewayClient.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public async Task<AntennaDto> UpdateAntennaAsync(Guid id, UpdateAntennaDto dto)
+    {
+        using var response = await httpClient.PutAsJsonAsync($"antennas/{id}", dto);
+        if (response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            throw new KeyNotFoundException($"Antenna {id} not found");
+        if (!response.IsSuccessStatusCode)
+            throw new GatewayException((int)response.StatusCode);
+        return await response.Content.ReadFromJsonAsync<AntennaDto>()
+            ?? throw new GatewayException(500);
+    }
```
Comment (BrunoPanizzi) [inline Backend.API/Features/Antennas/GatewayClient.cs:47]: O gateway retorna 204 ([No content](https://http.cat/status/204)) e aqui é esperado um retorno no body da resposta. O parse vai falhar mesmo que a operação tenha funcionado

File: Backend.API/Features/Antennas/GatewayClient.cs
Comment (BrunoPanizzi) [inline Backend.API/Features/Antennas/GatewayClient.cs:1]: Eu moveria todo código que toca o gateway para uma pasta fora de `Backend.API/Features`, talvez `Backend.API/Integrations`. O gateway não é uma funcionalidade do nosso backend, apenas é consumido por ele, então não faz sentido deixá-lo em features

File: Backend.API/Features/Antennas/AntennaService.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    public async Task<AntennaDto> UpdateAntennaAsync(Guid id, UpdateAntennaDto dto)
+    {
+        dto.Status ??= "Off";
+
+        if (dto.Sensibility.HasValue && (dto.Sensibility < 0 || dto.Sensibility > 100))
+            throw new ArgumentOutOfRangeException(nameof(dto.Sensibility), "Sensibility must be between 0 and 100");
+
+        if (dto.Power.HasValue && (dto.Power < 0 || dto.Power > 100))
+            throw new ArgumentOutOfRangeException(nameof(dto.Power), "Power must be between 0 and 100");
```
Comment (BrunoPanizzi) [inline Backend.API/Features/Antennas/AntennaService.cs:26]: Essas validações podem ser feitas em DTOs

**Review by BrunoPanizzi — CHANGES_REQUESTED** [review]

**Comment by BrunoPanizzi** [comment]
fechando pq a integração com antenas vai ser refatorada
